test/e2e/app-dir/dynamic-io/dynamic-io.test.ts

```diff
@@ -330,4 +331,46 @@ describe('dynamic-io', () => {
       expect($('#page-children').text()).toBe('at buildtime')
     }
   })
+
+  it('should not resume when client components are dynamic but the RSC render was static', async () => {
+    let html = await next.render('/cases/static-rsc-dynamic-client', {})
+    const $ = cheerio.load(html)
+
+    // Confirm the HTML document was sent completely
+    expect(html).toContain('</body></html>')
+
+    if (isNextDev) {
+      expect($('#layout').text()).toBe('at runtime')
+      expect($('#page').text()).toBe('at runtime')
+      // In dev we SSR the time
+      expect($('#time').length).toBe(1)
+    } else {
+      expect($('#layout').text()).toBe('at buildtime')
+      expect($('#page').text()).toBe('at buildtime')
+      // Confirm the time span is not part of the completed HTML document
+      expect($('#time').length).toBe(0)
+    }
+
+    const browser = await next.browser('/cases/static-rsc-dynamic-client')
+
+    const now = new Date()
+
+    if (isNextDev) {
+      expect(await browser.elementById('layout').text()).toBe('at runtime')
+      expect(await browser.elementById('page').text()).toBe('at runtime')
+      // Assert that we rendered a time within the last couple seconds.
+      const inPageDate = new Date(
+        await browser.waitForElementByCss('#time').text()
+      )
+      expect(inPageDate.getTime() - now.getTime()).toBeLessThan(2000)
+    } else {
+      expect(await browser.elementById('layout').text()).toBe('at buildtime')
+      expect(await browser.elementById('page').text()).toBe('at buildtime')
+      // Assert that we rendered a time within the last 5 seconds.
```

Comment:
I'm definitely not blocking on this, but it will make my eye twitch if you merge it like this. 😆 
```suggestion
      // Assert that we rendered a time within the last 2 seconds.
```